### PR TITLE
`azurerm_cosmosdb_postgresql_cluster`: add server names information t…

### PR DESCRIPTION
**Pull request by reastyn**
I have tried to deploy CosmosDB Postgres, but with some of the updates they added a random hexadecimal string to coordinator domain (from c-acctestcluster240313222245905652.postgres.cosmos.azure.com -> c-acctestcluster240313222245905652.**efyg2q5iqomwly**.postgres.cosmos.azure.com). When I wanted to use this domain to be passed to other modules I couldn't as I can derive the domain from the service name. Therefore I just propagated the server_names property to the attributes, which means it can be used as for example a output.

## Testing Logs/Evidence

I added the attribute to the basic test as seen in internal/services/cosmos/cosmosdb_postgresql_cluster_resource_test.go and successfully ran test locally.

## Change Log

* `azurerm_cosmosdb_postgresql_cluster` - add fully qualified domain name to attributes

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

### CODE REVIEW

**Comment by reastyn** [comment]
I also just noticed, that this stackoverflow mentions this problem 
https://stackoverflow.com/questions/76579820/output-azure-cosmos-db-for-postgresql-connection-string-with-terraform

File: internal/services/cosmos/cosmosdb_postgresql_cluster_resource.go [hunk 1]
```diff
@@ -636,3 +659,19 @@ func flattenMaintenanceWindow(input *clusters.MaintenanceWindow) []MaintenanceWi
 		},
 	}
 }
+
+func formatServerNames(input *[]clusters.ServerNameItem) []ServerNameItem {
```
Comment (stephybun) [inline internal/services/cosmos/cosmosdb_postgresql_cluster_resource.go:663]: Minor but for consistency could we rename this to 
```suggestion
func flattenServerNames(input *[]clusters.ServerNameItem) []ServerNameItem {
```

File: internal/services/cosmos/cosmosdb_postgresql_cluster_resource.go [hunk 2]
```diff
@@ -636,3 +659,19 @@ func flattenMaintenanceWindow(input *clusters.MaintenanceWindow) []MaintenanceWi
...
+	if input == nil {
+		return nil
+	}
+
+	var output []ServerNameItem
```
Comment (stephybun) [inline internal/services/cosmos/cosmosdb_postgresql_cluster_resource.go:668]: We should be setting an empty value into state instead of nil here 
```suggestion
	var output []ServerNameItem
	if input == nil {
		return output
	}

```

File: internal/services/cosmos/cosmosdb_postgresql_cluster_resource.go [hunk 3]
```diff
@@ -304,6 +310,22 @@ func (r CosmosDbPostgreSQLClusterResource) Attributes() map[string]*pluginsdk.Sc
 			Type:     pluginsdk.TypeString,
 			Computed: true,
 		},
+		"server_names": {
```
Comment (stephybun) [inline internal/services/cosmos/cosmosdb_postgresql_cluster_resource.go:313]: We can shorten this to `servers`, `_names` is implicit and one of the attributes in the block is already called `name`
```suggestion
		"servers": {
```

File: website/docs/r/cosmosdb_postgresql_cluster.html.markdown [hunk 4]
```diff
@@ -99,6 +99,8 @@ In addition to the Arguments listed above - the following Attributes are exporte
 
 * `earliest_restore_time` - The earliest restore point time (ISO8601 format) for the Azure Cosmos DB for PostgreSQL Cluster.
 
+* `server_names` - The name of the servers and its fully qualified domain name
```
Comment (stephybun) [inline website/docs/r/cosmosdb_postgresql_cluster.html.markdown:102]: Since this is a block it should be 
```suggestion
* `servers` - A `servers` block as defined below.

---

A `servers` block exports the following:

* `fqdn` - The Fully Qualified Domain Name of the server.

* `name` - The name of the server.
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @reastyn. I left some comments around consistency, would you mind taking a look at those? Once they're addressed we can take another look through.

**Comment by reastyn** [comment]
Thanks a lot for taking a look, the issues you mentioned should be resolved

**Review by stephybun — APPROVED** [review]
Thanks @reastyn LGTM 👍

**Comment by NemSimpraga** [comment]
Thank you @reastyn ! I am currently doing a patch via `azapi` , which I do **NOT** like!
```
data "azapi_resource" "db" {
  name                   = azurerm_cosmosdb_postgresql_cluster.db.name
  parent_id              = azurerm_resource_group.main.id
  type                   = "Microsoft.DBforPostgreSQL/serverGroupsv2@2023-03-02-preview"
  response_export_values = ["properties.serverNames"]
}
```
